core/ir/opnd.h

```diff
@@ -2316,6 +2316,16 @@ DR_API
  */
 bool
 opnd_set_index_extend(opnd_t *opnd, dr_extend_type_t extend, bool scaled);
+
+DR_API
+/**
+ * Assumes \p opnd is a base+disp memory reference.
+ * Sets the index register to be post-indexed.
```

Comment:
See the discussion above where this really does not change the semantic behavior: all it does is pick the post-indexed encoding for the case of a 0 displacement+immediate where the two encodings are semantically the same, for (rare) cases where someone cares about the encoding.  The comments should make that clear.  @AssadHashmi if you could confirm my interpretation of the `pre_index` field.

Reply:
Yes, as above pre/post index are taken care of by the codec.